src/modules/renderPage.ts

```diff
@@ -2,12 +2,15 @@ import { Location, updateHistoryRecord, getCurrentUrl } from '../helpers';
 import Swup from '../Swup';
 import { PageRecord } from './Cache';
 
+export type PageRenderOptions = {
+	event?: Event;
```

Comment:
`PopStateEvent` seems like a more exact option here. Could you explain why the generalization of the type here?

Reply:
While rethinking the types for the `loadPage` method without a `popstate` param, I thought about how useful it would be to have a reference to **any** type of event that triggered a visit. Right now it's just popstate events, but I can see lots of valid use cases for having access to the click and formsubmit events as well and passing them along to swup event handlers. That kind of matches the intention of the `triggerEvent` method, that passes along an `originalEvent` and not a `popstateEvent`. Does that make sense?

Reply:
Not sure about it. The consumer will always have to cast the event in handler to `as PopStateEvent` for example, to be able to read that it has a `state` on it, but I guess that's the case even now since multiple events are possible.  
Anyways, let's go with it! 👍

Reply:
True, right now it's a mouse or a popstate event, so the casting is already necessary.